cluster-operator/src/test/java/io/strimzi/operator/cluster/model/ZookeeperClusterTest.java

```diff
@@ -273,12 +274,52 @@ public void testCreateClusterWithZookeeperJmxEnabled() {
                 .build();
 
         ZookeeperCluster zookeeperCluster = ZookeeperCluster.fromCrd(Reconciliation.DUMMY_RECONCILIATION, kafka, KafkaVersionTestUtils.getKafkaVersionLookup());
-        Secret jmxSecret = zookeeperCluster.generateJmxSecret();
+        Secret jmxSecret = zookeeperCluster.generateJmxSecret(emptyMap(), emptyMap());
 
         assertThat(jmxSecret.getData(), hasKey("jmx-username"));
         assertThat(jmxSecret.getData(), hasKey("jmx-password"));
     }
 
+    @ParallelTest
+    public void testJmxSecretCustomLabelsAndAnnotations() {
+        Map<String, String> customLabels = new HashMap<>(2);
+        customLabels.put("label1", "value1");
+        customLabels.put("label2", "value2");
+
+        Map<String, String> customAnnotations = new HashMap<>(2);
+        customAnnotations.put("anno1", "value3");
+        customAnnotations.put("anno2", "value4");
+
+        Kafka kafka = new KafkaBuilder(ka)
+                .editSpec()
+                .editZookeeper()
+                .withJmxOptions(new KafkaJmxOptionsBuilder()
+                        .withAuthentication(new KafkaJmxAuthenticationPasswordBuilder()
+                                .build())
+                        .build())
+                .withNewTemplate()
+                .withNewJmxSecret()
+                .withNewMetadata()
+                .withAnnotations(customAnnotations)
+                .withLabels(customLabels)
+                .endMetadata()
+                .endJmxSecret()
+                .endTemplate()
+                .endZookeeper()
+                .endSpec()
+                .build();
```

Comment:
The code using the builder (here and in all other places) ismuch more readable when you use indentation to separate the different `withNew...` and `end...` calls. Have a look at the test below this one for example.